WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleCardPresentPaymentInLineMessage.swift

```diff
@@ -15,20 +15,20 @@ struct PointOfSaleCardPresentPaymentInLineMessage: View {
             PointOfSaleCardPresentPaymentPreparingForPaymentMessageView(viewModel: viewModel)
         case .tapSwipeOrInsertCard(let viewModel):
             PointOfSaleCardPresentPaymentTapSwipeInsertCardMessageView(viewModel: viewModel)
-        case .processing:
-            Text("processing...")
+        case .processing(let viewModel):
+            POSCardPresentPaymentMessageView(viewModel: POSCardPresentPaymentMessageViewModel(imageName: viewModel.imageName, title: viewModel.title))
         case .displayReaderMessage(let viewModel):
             PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView(viewModel: viewModel)
-        case .paymentSuccess:
-            Text("Payment successful!")
+        case .paymentSuccess(let viewModel):
+            POSCardPresentPaymentMessageView(viewModel: POSCardPresentPaymentMessageViewModel(imageName: viewModel.imageName, title: viewModel.title))
```

Comment:
https://github.com/woocommerce/woocommerce-ios/assets/4062343/c4a9568d-70ec-43d3-b941-88e2c8f04439

It may not be related to these changes but I noticed that the success view is briefly visible when starting another payment. Likely `cardPresentPaymentInlineMessage` is not reset. If `TotalsViewModel` _holds_ a state for a specific payment, we could consider creating a new one every time a new payment starts and avoid such a category of issues. Let me know if I should create an issue for it.

Reply:
Yeah, I noticed that few days ago. Created https://github.com/woocommerce/woocommerce-ios/issues/13275